pennylane/transforms/unitary_to_rot.py

```diff
@@ -23,7 +24,66 @@
 from pennylane.typing import PostprocessingFn
 
 
-@transform
+@lru_cache
+def _get_plxpr_unitary_to_rot():
+    try:
+        # pylint: disable=import-outside-toplevel
+        from jax import make_jaxpr
+
+        from pennylane.capture import PlxprInterpreter
+        from pennylane.operation import Operator
+    except ImportError:  # pragma: no cover
+        return None, None
+
+    # pylint: disable=redefined-outer-name, too-few-public-methods
+    class UnitaryToRotInterpreter(PlxprInterpreter):
+        """Plxpr Interpreter for applying the ``unitary_to_rot``
+        transform when program capture is enabled."""
+
+        def interpret_operation(self, op: Operator):
+            """Decompose a PennyLane operation instance if it is a QubitUnitary.
+
+            Args:
+                op (Operator): a pennylane operator instance
+
+            Returns:
+                list: The decomposed operations.
+
+            This method is only called when the operator's output is a dropped variable,
+            so the output will not affect later equations in the circuit.
+
+            See also: :meth:`~.interpret_operation_eqn`, :meth:`~.interpret_operation`.
+            """
+            if isinstance(op, qml.QubitUnitary):
+                ops = []
+                with qml.capture.pause():
+                    matrix_shape = qml.math.shape(op.parameters[0])
+                    if matrix_shape == (2, 2):
+                        ops = one_qubit_decomposition(op.parameters[0], op.wires[0])
+                    elif matrix_shape == (4, 4):
+                        ops = two_qubit_decomposition(op.parameters[0], op.wires)
+                return super().interpret_operation(ops)
```

Comment:
`interpret_operation` is for single ops, not sequences.
```suggestion
                return [super().interpret_operation(o) for o in ops]
```